algviz/picture/connection_map.py

```diff
@@ -0,0 +1,136 @@
+import svgutils.transform as svgutils
+import os
+
+from algviz.parser import structures
+from .anchor import Anchor
+from .svg_engine import SVGEngine
+
+"""Tracks connection points and connection pictures to be composed into a single picture"""
+
+pointers_temp_file = "temp_pointers.svg"
+
+class ConnectionMap:
+    class Connection:
+        def __init__(self, start_point, connect_pic, anchor):
+            self.start_point = start_point
+            self.connect_pic = connect_pic
+            if not connect_pic.is_drawn:
+                connect_pic.draw()
+            self.anchor = anchor
+
+            self.left_shift_y = 0
+            self.right_shift_y = 0
+
+        def is_pointer(self):
+            return False
+
+    class PointerConnection(Connection):
+        def __init__(self, start_point, connect_pic, anchor, pointer_style={}):
+            self.pointer_style = pointer_style
+            super().__init__(self, start_point, connect_pic, anchor)
+
+        def is_pointer(self):
+            return True
+    
+    def __init__(self, startPicture):
+        self.connections = []
+        self.picture = startPicture
+        
+    def add_connection(self, point, connect_pic, anchor=None, pointer=True, pointer_style={}):
+        if self.picture.width < point[0] or self.picture.height < point[1]:
+            raise TypeError("Point not included in picture")
```

Comment:
Should be a `ValueError`, not a `TypeError`.  (I mixed these up myself until recently.)